index.js

```diff
@@ -72,22 +71,25 @@ function findAllDependencies(file, knownDependencies, sourceDirectories, knownFi
 }
 
 // Given a source directory (containing top-level Elm modules), locate the
-// elm-package.json file that includes it.
-function findElmPackage(baseDir, currentDir) {
+// elm-package.json file that includes it and get all its source directories.
+function getElmPackageSourceDirectories(baseDir, currentDir) {
   if (typeof currentDir === "undefined") {
     currentDir = baseDir;
   }
 
   var elmPackagePath = path.join(currentDir, 'elm-package.json');
   if (fs.existsSync(elmPackagePath)) {
-    return elmPackagePath;
+    var sourceDirectories = getSourceDirectories(elmPackagePath);
+    if (_.includes(sourceDirectories, baseDir)) {
```

Comment:
It will halt when the search for a matching elm-package.json reaches the root directory of the file system (see `return false;` on line 89).

Basically we ignore the “bad” elm-package.json file and continue searching up the path for a “good” elm-package.json file. If we never find one, we give up when we get to the filesystem root.

Hope that makes sense!